packages/@lwc/engine-dom/src/renderer/index.ts

```diff
@@ -0,0 +1,346 @@
+    target.removeEventListener(type, callback, options);
+}
+
+function dispatchEvent(target: Node, event: Event): boolean {
+    return target.dispatchEvent(event);
+}
+
+function getClassList(element: Element): DOMTokenList {
+    return element.classList;
+}
+
+function setCSSStyleProperty(
+    element: Element,
+    name: string,
+    value: string,
+    important: boolean
+): void {
+    // TODO [#0]: How to avoid this type casting? Shall we use a different type interface to
+    // represent elements in the engine?
+    (element as HTMLElement | SVGElement).style.setProperty(
+        name,
+        value,
+        important ? 'important' : ''
+    );
+}
+
+function getBoundingClientRect(element: Element): DOMRect {
+    return element.getBoundingClientRect();
+}
+
+function querySelector(element: Element, selectors: string): Element | null {
+    return element.querySelector(selectors);
+}
+
+function querySelectorAll(element: Element, selectors: string): NodeList {
+    return element.querySelectorAll(selectors);
+}
+
+function getElementsByTagName(element: Element, tagNameOrWildCard: string): HTMLCollection {
+    return element.getElementsByTagName(tagNameOrWildCard);
+}
+
+function getElementsByClassName(element: Element, names: string): HTMLCollection {
+    return element.getElementsByClassName(names);
+}
+
+function getChildren(element: Element): HTMLCollection {
+    return element.children;
+}
+
+function getChildNodes(element: Element): NodeList {
+    return element.childNodes;
+}
+
+function getFirstChild(element: Element): Node | null {
+    return element.firstChild;
+}
+
+function getFirstElementChild(element: Element): Element | null {
+    return element.firstElementChild;
+}
+
+function getLastChild(element: Element): Node | null {
+    return element.lastChild;
+}
+
+function getLastElementChild(element: Element): Element | null {
+    return element.lastElementChild;
+}
+
+function isConnected(node: Node): boolean {
+    return node.isConnected;
+}
+
+function assertInstanceOfHTMLElement(elm: any, msg: string) {
+    assert.invariant(elm instanceof HTMLElement, msg);
+}
+
+const HTMLElementExported = HTMLElementConstructor as typeof HTMLElement;
+
+export {
+    HTMLElementExported,
+    insert,
+    remove,
+    cloneNode,
+    createFragment,
+    createElement,
+    createText,
+    createComment,
+    nextSibling,
+    attachShadow,
+    getProperty,
+    setProperty,
+    setText,
+    getAttribute,
+    setAttribute,
+    removeAttribute,
+    addEventListener,
+    removeEventListener,
+    dispatchEvent,
+    getClassList,
+    setCSSStyleProperty,
+    getBoundingClientRect,
+    querySelector,
+    querySelectorAll,
+    getElementsByTagName,
+    getElementsByClassName,
+    getChildren,
+    getChildNodes,
+    getFirstChild,
+    getFirstElementChild,
+    getLastChild,
+    getLastElementChild,
+    isConnected,
+    assertInstanceOfHTMLElement,
+    defineCustomElement,
+    getCustomElement,
+};
```

Comment:
I could have proved my point by splitting this up into multiple files (e.g. `parseFragment`), but I'm afraid of bitrotting myself if this PR takes a long time to get reviewed and merged.

So I mostly just copy-pasted. The only change I made was to rely on `assert`, `isNull`, and `isUndefined` from `@lwc/shared`.